packages/core/diagnostic/src/diagnostic.js

```diff
@@ -4,49 +4,71 @@ import type {FilePath} from '@parcel/types';
 export type DiagnosticHighlightLocation = {|
-  // These positions are 1-based
   +line: number,
   +column: number,
 |};
 
 export type DiagnosticSeverity = 'error' | 'warn' | 'info';
 
-// Note: A tab character is always counted as a single character
-// This is to prevent any mismatch of highlighting across machines
+/**
+ * Note: A tab character is always counted as a single character
+ * This is to prevent any mismatch of highlighting across machines
+ */
 export type DiagnosticCodeHighlight = {|
-  // start and end are included in the highlighted region
+  /** Location of the first character that should get highlighted for this highlight. */
   start: DiagnosticHighlightLocation,
+  /** Location of the last character that should get highlighted for this highlight. */
   end: DiagnosticHighlightLocation,
+  /** A message that should be displayed at this location in the code (optional). */
   message?: string,
 |};
 
+/**
+ * Describes how to format a code frame.
+ * A code frame is a visualization of a piece of code with a certain amount of
+ * code highlights that point to certain chunk(s) inside the code.
+ */
 export type DiagnosticCodeFrame = {|
-  // if no code is passed, it will be read in from Diagnostic#filePath
+  /**
+   * The contents of the source file.
+   *
+   * If no code is passed, it will be read in from Diagnostic#filePath, remember that
+   * the asset's current code could be different from the input contents.
+   *
+   */
   code?: string,
   codeHighlights: DiagnosticCodeHighlight | Array<DiagnosticCodeHighlight>,
 |};
 
-// A Diagnostic is a style agnostic way of emitting errors, warnings and info
-// The reporter's are responsible for rendering the message, codeframes, hints, ...
+/**
+ * A style agnostic way of emitting errors, warnings and info.
+ * Reporters are responsible for rendering the message, codeframes, hints, ...
+ */
 export type Diagnostic = {|
+  /** This is the message you want to log. */
   message: string,
-  origin?: string, // Name of plugin or file that threw this error
+  /** Name of plugin or file that threw this error */
+  origin?: string,
 
-  // basic error data
+  /** A stacktrace of the error (optional) */
   stack?: string,
+  /** Name of the error (optional) */
   name?: string,
 
-  // Asset metadata, filePath is absolute or relative to the project root
+  /** Path to the file this diagnostic is about (optional, absolute or relative to the project root) */
   filePath?: FilePath,
+  /** Language of the file this diagnostic is about (optional) */
   language?: string,
 
-  // Codeframe data
+  /** A code frame points to a certain location(s) in the file this diagnostic is linked to (optional) */
   codeFrame?: DiagnosticCodeFrame,
 
-  // Hints to resolve issues faster
+  /** An optional list of strings that suggest ways to resolve this issue */
   hints?: Array<string>,
 
+  /** TODO? */
   skipFormatting?: boolean,
```

Comment:
...

Reply:
This should actually be hidden in the docs, we only really use it to not format things that weren't originally designed for diagnostics